devices/Button/Tests/ButtonTests.cs

```diff
@@ -0,0 +1,219 @@
+using Iot.Device.Tests;
+using nanoFramework.TestFramework;
+using System.Threading;
+
+namespace Tests
+{
+    [TestClass]
+    class Buttontests
+    {
+        [TestMethod]
+        public void If_Button_Is_Once_Pressed_Press_Event_Fires()
+        {
+            bool Pressed = false;
+            bool Holding = false;
+            bool DoublePressed = false;
+
+            TestButton button = new TestButton();
+
+            button.Press += (sender, e) =>
+            {
+                Pressed = true;
+            };
+
+            button.Holding += (sender, e) =>
+            {
+                Holding = true;
+            };
+
+            button.DoublePress += (sender, e) =>
+            {
+                DoublePressed = true;
+            };
+
+            button.PressButton();
+            Thread.Sleep(100);
```

Comment:
a little comment on why the Thread.Sleep in all this file will be more than welcome

Reply:
Great suggestion, should be added now.